libs/datamodel/core/src/transform/ast_to_dml/db/walkers/relation.rs

```diff
@@ -369,18 +368,7 @@ impl<'ast, 'db> CompleteInlineRelationWalker<'ast, 'db> {
     pub(crate) fn on_update(self) -> ReferentialAction {
         use ReferentialAction::*;
 
-        self.referencing_field().attributes().on_update.unwrap_or_else(|| {
-            let uses_foreign_keys = self
-                .db
-                .active_connector()
-                .has_capability(ConnectorCapability::ForeignKeys);
-
-            match self.referential_arity() {
-                _ if uses_foreign_keys => Cascade,
-                ast::FieldArity::Required => NoAction,
-                _ => SetNull,
-            }
-        })
+        self.referencing_field().attributes().on_update.unwrap_or(Cascade)
```

Comment:
This will probably clash with your PR Dom but I had to remove the references to `ConnectorCapability::ForeignKeys` since we removed it